<h1 align="center">Thesis-Data Collection</h1>
<p align="left">A repository for the data collection stage of my degree project in protein truncation.</p><br>

[**Relevant Databases**](#RDB)
+ [**N- & C-Terminal variants**](#VariantNC)
+ [**Variants in the middle of sequences originating from exon skipping**](#VariantM)
+ [**Disease related (and not) variants**](#Disease)

[**Existing methods for the prediction of the outcome of variants**](#MethodPred)

<br><a name="RDB"></a>
<h2 align="center">Relevant DataBases</h2>

<a name="VariantNC"></a>
## N- & C-Terminal variants
###
[**TopFind**](https://topfind.clip.msl.ubc.ca/)
+ [Lange PF, Huesgen PF, Overall CM. TopFIND 2.0--linking protein termini with proteolytic processing and modifications altering protein function. Nucleic Acids Res. 2012;40(Database issue):D351-D361](https://www.researchgate.net/publication/51552726_TopFIND_a_knowledgebase_linking_protein_termini_with_function)

<a name="VariantM"></a>
## Variants in the middle of sequences originating from exon skipping
###
[**ExonSkipDB**](https://ccsm.uth.edu/ExonSkipDB/)
+ [Pora Kim, Mengyuan Yang, Ke Yiya, Weiling Zhao, Xiaobo Zhou, ExonSkipDB: functional annotation of exon skipping event in human, Nucleic Acids Research, Volume 48, Issue D1, 08 January 2020, Pages D896–D907](https://academic.oup.com/nar/article/48/D1/D896/5603217#190995057)

[**ASpedia**](http://combio.snu.ac.kr/aspedia/index.html)
+ [Daejin Hyung, Jihyun Kim, Soo Young Cho, Charny Park, ASpedia: a comprehensive encyclopedia of human alternative splicing, Nucleic Acids Research, Volume 46, Issue D1, 4 January 2018, Pages D58–D63](https://academic.oup.com/nar/article/46/D1/D58/4584641)

<a name="Disease"></a>
## Disease related (and not) variants
###
[**ClinVar**](https://www.ncbi.nlm.nih.gov/clinvar/)
+ [Eduardo Pérez-Palma, Marie Gramm, Peter Nürnberg, Patrick May, Dennis Lal, Simple ClinVar: an interactive web server to explore and retrieve gene and disease variants aggregated in ClinVar database, Nucleic Acids Research, Volume 47, Issue W1, 02 July 2019, Pages W99–W105](https://academic.oup.com/nar/article/47/W1/W99/5494761)
+ [Landrum MJ, Lee JM, Riley GR, et al. ClinVar: public archive of relationships among sequence variation and human phenotype. Nucleic Acids Res. 2014;42(Database issue):D980-D985](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3965032/)

<a name="MethodPred"></a>
<br><h2 align="center">Existing methods for the prediction of the outcome of variants</h2>

###
[**MutPred-Indel**](http://mutpredindel.cs.indiana.edu/)
+ [Pagel KA, Antaki D, Lian A, et al. Pathogenicity and functional impact of non-frameshifting insertion/deletion variation in the human genome. PLoS Comput Biol. 2019;15(6):e1007112. Published 2019 Jun 14](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3965032/)

[**SIFT-Indel**](https://sift.bii.a-star.edu.sg/www/SIFT_indels2.html)
+ [Hu J, Ng PC. SIFT Indel: predictions for the functional effects of amino acid insertions/deletions in proteins. PLoS One. 2013;8(10):e77940. Published 2013 Oct 23](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3806772/)
