<h1 align="center">BINP39 Project</h1>
<p align="left">The repository of my master's BINP39 project in <em>Variant Visualization</em> for <a href="http://structure.bmc.lu.se/PON-SC">PON-SC</a> at Lund University.</p>
<p>Project tutor: <em>Mauno Vihinen</em></p><br>

[*Relevant Tools*](#RTool)
+ [**Secondary Structural Identification**](#STRIDE)
+ [**A Backbone-dependent Rotamer Library**](#ROTAMER)

[*Existing papers of relevant analysis and algorithms*](#papers)
+ [**A meta-analysis of nonsense mutations causing human genetic disease <em>(read)</em>**](#MetaAnalysis)

[*Existing methods for the prediction of the outcome of variants*](#MethodPred)
+ [**MutPred-Indel**](#mutpred-indel)

[*Tutorials of potentially useful tools for data collection*](#tutorial)
+ [**Jupyter Notebook**](#jupyter)
+ [**Scrapy**](#scrapy)
+ [**Pandas**](#pandas)
+ [**BioPython**](#biopython)
+ [**ScikitLearn**](#scikitlearn)

<br><a name="RTool"></a>
<h2 align="center">Relevant DataBases</h2>

<a name="STRIDE"></a>
## Secondary Structural Identification
###
[**STRIDE**](http://webclu.bio.wzw.tum.de/stride/)
+ [Heinig M, Frishman D. STRIDE: a web server for secondary structure assignment from known atomic coordinates of proteins. Nucleic Acids Res. 2004;32(Web Server issue):W500-W502. doi:10.1093/nar/gkh429](https://pubmed.ncbi.nlm.nih.gov/15215436/)

<a name="ROTAMER"></a>
## Rotamer Library
###
[**A Backbone-dependent Rotamer Library**](http://dunbrack.fccc.edu/bbdep2010/)
+ [Shapovalov MV, Dunbrack RL Jr. A smoothed backbone-dependent rotamer library for proteins derived from adaptive kernel density estimates and regressions. Structure. 2011;19(6):844-858. doi:10.1016/j.str.2011.03.019](http://dunbrack.fccc.edu/bbdep2010/)


<a name="MethodPred"></a>
<br><h2 align="center">Existing methods for the prediction of the outcome of variants</h2>

###
<a name="mutpred-indel"></a>
[**MutPred-Indel**](http://mutpredindel.cs.indiana.edu/)
+ [Mort, M., Sterne-Weiler, T., Li, B. et al. MutPred Splice: machine learning-based prediction of exonic variants that disrupt splicing. Genome Biol 15, R19 (2014)](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-1-r19#citeas)

<a name="papers"></a>
<br><h2 align="center">Existing papers of relevant analysis and algorithms</h2>

###
<a name="MetaAnalysis"></a>
[**Mort M, Ivanov D, Cooper DN, Chuzhanova NA. A meta-analysis of nonsense mutations causing human genetic disease. Hum Mutat. 2008;29(8):1037-1047.**](https://pubmed.ncbi.nlm.nih.gov/18454449/)

<a name="tutorial"></a>
<br><h2 align="center">Tutorials of potentially useful tools for data collection</h2>

###
<a name="jupyter"></a>
[**Jupyter Notebook**](https://jupyter.org/)
+ [Jupyter Notebook Tutorial: The Definitive Guide](https://www.datacamp.com/community/tutorials/tutorial-jupyter-notebook)
+ [Jupyter Notebook for Beginners: A Tutorial towards .ipynb](https://www.dataquest.io/blog/jupyter-notebook-tutorial/)

<a name="scrapy"></a>
[**Scrapy**](https://docs.scrapy.org/en/latest/intro/tutorial.html)
+ [Python Scrapy tutorial for beginners – 01 – Creating your first spider](https://letslearnabout.net/tutorial/scrapy-tutorial/python-scrapy-tutorial-for-beginners-01-creating-your-first-spider/)
+ [Python Scrapy tutorial for beginners – 02 – Extract all the data!](https://letslearnabout.net/tutorial/python-scrapy-tutorial-for-beginners-02-extract-all-the-data/)

<a name="pandas"></a>
[**Pandas**](https://pandas.pydata.org/docs/)
+ [Selecting Subsets of Data in Pandas: Part 1](https://medium.com/dunder-data/selecting-subsets-of-data-in-pandas-6fcd0170be9c)
+ [How to select rows from a DataFrame based on column values?](https://stackoverflow.com/questions/17071871/how-to-select-rows-from-a-dataframe-based-on-column-values)

<a name="biopython"></a>
[**BioPython**](https://biopython.org/)
+ [Package PDB](https://biopython.org/DIST/docs/api/Bio.PDB-module.html)

<a name="scikitlearn"></a>
[**ScikitLearn**](https://scikit-learn.org/stable/)
+ [neighbours.KDTree](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KDTree.html)
