<h1 align="center">BINP39 Project</h1>
<p align="left">The repository of my master's BINP39 project in <em>Variant Visualization</em> for <a href="http://structure.bmc.lu.se/PON-SC">PON-SC</a> at Lund University.</p>
<p>Project tutor: <em>Mauno Vihinen</em></p><br>

[*Relevant Tools*](#RTool)
+ [**Secondary Structural Identification**](#STRIDE)
+ [**A Backbone-dependent Rotamer Library**](#ROTAMER)
+ [**JSmol, the web version of a 3D chemical structure viewer**](#jsmol)
+ [**3Dmol.js: molecular visualization with WebGL**](#3dmol)
+ [**VarSite: Disease variants and protein structure**](#varsite)
+ [**HUMA: A platform for the analysis of genetic variation in humans**](#huma)
+ [**BTKbase: the mutation database for X-linked agammaglobulinemia**](#btkbase)

[*Existing papers of relevant analysis and algorithms*](#papers)
+ [**PON-SC – program for identifying steric clashes caused by amino acid substitutions <em>(read)</em>**](#ponsc)

[*Tutorials of potentially useful software and packages*](#tutorial)
+ [**Jupyter Notebook**](#jupyter)
+ [**Scrapy**](#scrapy)
+ [**Pandas**](#pandas)
+ [**BioPython**](#biopython)
+ [**ScikitLearn**](#scikitlearn)

<br><a name="RTool"></a>
<h2 align="center">Relevant Tools</h2>

<a name="STRIDE"></a>
## Secondary Structural Identification
###
[**STRIDE**](http://webclu.bio.wzw.tum.de/stride/)
+ [Heinig M, Frishman D. STRIDE: a web server for secondary structure assignment from known atomic coordinates of proteins. Nucleic Acids Res. 2004;32(Web Server issue):W500-W502. doi:10.1093/nar/gkh429](https://pubmed.ncbi.nlm.nih.gov/15215436/)

<a name="ROTAMER"></a>
## Rotamer Library
###
[**A Backbone-dependent Rotamer Library**](http://dunbrack.fccc.edu/bbdep2010/)
+ [Shapovalov MV, Dunbrack RL Jr. A smoothed backbone-dependent rotamer library for proteins derived from adaptive kernel density estimates and regressions. Structure. 2011;19(6):844-858. doi:10.1016/j.str.2011.03.019](https://pubmed.ncbi.nlm.nih.gov/21645855/)

<a name="jsmol"></a>
## JSmol
###
[**An open-source html5-modality viewer for chemical structures in 3D**](https://sourceforge.net/projects/jsmol/)
+ [Official Tutorial](http://wiki.jmol.org/index.php/JSmol)
+ [Additional Tutorial 1](https://chemapps.stolaf.edu/jmol/jsmol/jsmol.htm)
+ [Addtional Tutorial 2](https://www.andrew.cmu.edu/user/rule/jsmol/jsmol_tutorial.html)
+ [Simple web demo in JavaScript](http://jmol.sourceforge.net/demo/jssample0.html)

<a name="3dmol"></a>
## 3Dmol
###
[**3Dmol.js: molecular visualization with WebGL**](https://3dmol.csb.pitt.edu/)
+ [Github Link](https://github.com/3dmol/3Dmol.js)
+ [Rego N, Koes D. 3Dmol.js: molecular visualization with WebGL. Bioinformatics. 2015;31(8):1322-1324. doi:10.1093/bioinformatics/btu829](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4393526/)

<a name="varsite"></a>
## Varsite
###
[**VarSite: Disease variants and protein structure**](https://www.ebi.ac.uk/thornton-srv/databases/VarSite)
+ [Laskowski RA, Stephenson JD, Sillitoe I, Orengo CA, Thornton JM. VarSite: Disease variants and protein structure. Protein Sci. 2020;29(1):111-119. doi:10.1002/pro.3746](https://pubmed.ncbi.nlm.nih.gov/31606900/)

<a name="huma"></a>
## HUMA
###
[**HUMA: A platform for the analysis of genetic variation in humans**](https://huma.rubi.ru.ac.za)
+ [Brown DK, Tastan Bishop Ö. HUMA: A platform for the analysis of genetic variation in humans. Hum Mutat. 2018;39(1):40-51. doi:10.1002/humu.23334](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5722678/)

<a name="btkbase"></a>
## BTKbase
###
[**BTKbase: the mutation database for X-linked agammaglobulinemia**](http://structure.bmc.lu.se/idbase/BTKbase/index.php?content=bioinfor/IDbases)
+ [Väliaho J, Smith CI, Vihinen M. BTKbase: the mutation database for X-linked agammaglobulinemia. Hum Mutat. 2006;27(12):1209-1217. doi:10.1002/humu.20410
](https://pubmed.ncbi.nlm.nih.gov/16969761/)


<a name="papers"></a>
<br><h2 align="center">Existing papers of relevant analysis and algorithms</h2>

###
<a name="ponsc"></a>
[**Čalyševa J, Vihinen M. PON-SC - program for identifying steric clashes caused by amino acid substitutions. BMC Bioinformatics. 2017;18(1):531. Published 2017 Nov 29. doi:10.1186/s12859-017-1947-7**](https://pubmed.ncbi.nlm.nih.gov/29187139/)

<a name="tutorial"></a>
<br><h2 align="center">Tutorials of potentially useful software and packages</h2>

###
<a name="jupyter"></a>
[**Jupyter Notebook**](https://jupyter.org/)
+ [Jupyter Notebook Tutorial: The Definitive Guide](https://www.datacamp.com/community/tutorials/tutorial-jupyter-notebook)
+ [Jupyter Notebook for Beginners: A Tutorial towards .ipynb](https://www.dataquest.io/blog/jupyter-notebook-tutorial/)

<a name="scrapy"></a>
[**Scrapy**](https://docs.scrapy.org/en/latest/intro/tutorial.html)
+ [Python Scrapy tutorial for beginners – 01 – Creating your first spider](https://letslearnabout.net/tutorial/scrapy-tutorial/python-scrapy-tutorial-for-beginners-01-creating-your-first-spider/)
+ [Python Scrapy tutorial for beginners – 02 – Extract all the data!](https://letslearnabout.net/tutorial/python-scrapy-tutorial-for-beginners-02-extract-all-the-data/)

<a name="pandas"></a>
[**Pandas**](https://pandas.pydata.org/docs/)
+ [Selecting Subsets of Data in Pandas: Part 1](https://medium.com/dunder-data/selecting-subsets-of-data-in-pandas-6fcd0170be9c)
+ [How to select rows from a DataFrame based on column values?](https://stackoverflow.com/questions/17071871/how-to-select-rows-from-a-dataframe-based-on-column-values)

<a name="biopython"></a>
[**BioPython**](https://biopython.org/)
+ [Package PDB](https://biopython.org/DIST/docs/api/Bio.PDB-module.html)

<a name="scikitlearn"></a>
[**ScikitLearn**](https://scikit-learn.org/stable/)
+ [neighbours.KDTree](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KDTree.html)
